inc/ThirdParty/Themes/Avada.php

```diff
@@ -0,0 +1,124 @@
+	/**
+	 * Options instance
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.3.1
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		$current_theme = wp_get_theme();
+
+		if ( 'Avada' !== $current_theme->get( 'Name' ) ) {
+			return [];
+		}
+
+		return [
+			'avada_clear_dynamic_css_cache'        => 'clean_domain',
+			'rocket_exclude_defer_js'              => 'exclude_defer_js',
+			'rocket_maybe_disable_lazyload_helper' => 'maybe_disable_lazyload',
+			'fusion_cache_reset_after'             => 'clear_cache_fusion_patcher',
+			'update_option_fusion_options'         => [ 'maybe_deactivate_lazyload', 10, 2 ],
+		];
+	}
+
+	/**
+	 * Constructor
+	 *
+	 * @param Options_Data $options WP Rocket options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * When Avada theme purge its own cache.
+	 * Clear WP Rocket cache when Avada dynamic CSS is updated.
+	 *
+	 * @return void
+	 */
+	public function clean_domain() {
+		rocket_clean_domain();
+	}
+
+	/**
+	 * Deactivate WP Rocket lazyload if Avada lazyload is enabled.
+	 *
+	 * @since 3.3.4
+	 *
+	 * @param string $old_value Previous Avada option value.
+	 * @param string $value     New Avada option value.
+	 * @return void
+	 */
+	public function maybe_deactivate_lazyload( $old_value, $value ) {
+		if (
+			empty( $old_value['lazy_load'] )
+			||
+			( ! empty( $value['lazy_load'] ) && 'avada' === $value['lazy_load'] )
+		) {
+			update_rocket_option( 'lazyload', 0 );
+		}
+	}
+
+	/**
+	 * Excludes Avada Google Maps JS files from defer JS
+	 *
+	 * @param array $exclude_defer_js Array of JS filepaths to be excluded.
+	 * @return array
+	 */
+	public function exclude_defer_js( $exclude_defer_js ) {
+		$exclude_defer_js[] = '/jquery-?[0-9.]*(.min|.slim|.slim.min)?.js';
+		$exclude_defer_js[] = rocket_clean_exclude_file( 'maps.googleapis.com' );
+
+		return $exclude_defer_js;
+	}
+
+	/**
+	 * Disable WP Rocket lazyload field if Avada lazyload is enabled
+	 *
+	 * @since 3.3.4
+	 * @param  array $disable_images_lazyload Array with plugins which disable lazyload functionality.
+	 * @return array
+	 */
+	public function maybe_disable_lazyload( $disable_images_lazyload ) {
+		$avada_options = get_option( 'fusion_options' );
+
+		if ( empty( $avada_options['lazy_load'] ) ) {
+			return $disable_images_lazyload;
+		}
+
+		if ( ! empty( $avada_options['lazy_load'] && 'avada' !== $avada_options['lazy_load'] ) ) {
+			return $disable_images_lazyload;
+		}
+
+		$disable_images_lazyload[] = __( 'Avada', 'rocket' );
+		return $disable_images_lazyload;
+	}
+
+	/**
+	 * Clears WP Rocket's cache after Avada's Fusion Patcher flushes their caches
+	 *
+	 * @since 3.3.5
+	 *
+	 * @return void
+	 */
+	public function clear_cache_fusion_patcher() {
+		rocket_clean_domain();
+	}
```

Comment:
You have 2 methods doing the same thing in the class, that's not necessary, you can just re-use the first one